src/current/v23.2/plpgsql.md

```diff
@@ -0,0 +1,417 @@
+
+`IF ... THEN ... ELSE` executes statements if a boolean condition is true. If the condition is false, the `ELSE` statements are executed.
+
+~~~ sql
+IF condition THEN
+	statements;
+  ELSE
+	else_statements;
+  END IF;
+~~~
+
+`IF ... THEN ... ELSEIF` executes statements if a boolean condition is true. If the condition is false, each `ELSIF` condition is evaluated until one is true. The corresponding `ELSIF` statements are executed. If no `ELSIF` conditions are true, no statements are executed unless an `ELSE` clause is included, in which case the `ELSE` statements are executed.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+IF condition THEN
+	statements;
+  ELSIF elsif_condition THEN
+	elsif_statements;
+  [ ELSIF elsif_condition_n THEN
+	elsif_statements_n; ]
+  [ ELSE
+	else_statements; ]
+  END IF;
+~~~
+
+For usage examples of conditional statements, see [Examples](#examples).
+
+### Write loops
+
+Use looping syntax to repeatedly execute statements.
+
+On its own, `LOOP` executes statements infinitely.
+
+~~~ sql
+LOOP
+	statements;
+  END LOOP;
+~~~
+
+On its own, `WHILE` executes statements infinitely if a boolean condition is true. The statements repeat until the condition is false.
+
+~~~ sql
+WHILE condition LOOP
+	statements;
+  END LOOP;
+~~~
+
+For an example, see [Create a stored procedure that uses a `WHILE` loop]({% link {{ page.version.version }}/create-procedure.md %}#create-a-stored-procedure-that-uses-a-while-loop).
+
+Add an `EXIT` statement to end a `LOOP` or `WHILE` statement block. This should be combined with a [conditional statement](#write-conditional-statements).
+
+~~~ sql
+LOOP
+	statements;
+	IF condition THEN
+		EXIT;
+	END IF;
+  END LOOP;
+~~~
+
+Add a `CONTINUE` statement to end a `LOOP` or `WHILE` statement block, skipping any statements below `CONTINUE`, and begin the next iteration of the loop. This should be combined with a [conditional statement](#write-conditional-statements). In the following example, if the `IF` condition is met, then `CONTINUE` causes the loop to skip the second block of statements and begin again.
+
+~~~ sql
+LOOP
+	statements;
+	IF condition THEN
+		CONTINUE;
+	END IF;
+	statements;
+  END LOOP;
+~~~
+
+### Open and use cursors
+
+PL/pgSQL cursors can be used in the following scenarios:
+
+- When [declared as variables](#declare-cursor-variables), cursors can be used within [PL/pgSQL blocks](#structure).
+- When specified as a parameter in a [`CREATE PROCEDURE`]({% link {{ page.version.version }}/create-procedure.md %}) statement, cursors can be accessed externally from the stored procedure.
+
+The cursor must first be opened within a PL/pgSQL block. If the cursor was declared without being bound to a query, you must specify a query using the `FOR` clause.
+
+~~~ sql
+BEGIN
+	OPEN cursor_name [ FOR query ];
+	...
+~~~
+
+You can then use the `FETCH` statement to retrieve query results for a specified number of rows:
+
+~~~ sql
+FETCH FORWARD rows FROM cursor_name;
```

Comment:
We should clarify that this is a SQL FETCH statement, not a PL/pgSQL FETCH statement, and that the transaction where the cursor was opened has to be still be open as well (or maybe this last detail is below?).

Reply:
PL/pgSQL MOVE is basically just like SQL MOVE, but FETCH is a bit different:
* PL/pgSQL FETCH requires an INTO clause to assign the result to variables.
* PL/pgSQL FETCH only returns a single row. If using something like `FETCH 10`, this is equivalent to moving 10 rows and then returning the last.

Reply:
Thank you clarifying! Added this.